### PR TITLE
[backport] Update doctest of `cupyx.scipy.linalg.lu_factor`

### PR DESCRIPTION
**Pull request by kmaehashi**
Backport of #4561

### CODE REVIEW

**Comment by kmaehashi** [comment]
[automatic post] Jenkins, test this please.

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
LGTM! Waiting for CI pass.

File: tests/cupyx_tests/scipy_tests/linalg_tests/test_decomp_lu.py [hunk 1]
```diff
@@ -32,10 +33,9 @@ def test_lu_factor(self, dtype):
         cupy.testing.assert_allclose(result_cpu[0], result_gpu[0], atol=1e-5)
         cupy.testing.assert_array_equal(result_cpu[1], result_gpu[1])
 
-    @testing.for_float_dtypes(no_float16=True)
-    def test_lu_factor_reconstruction(self, dtype):
+    @testing.for_dtypes('fdFD')
```
Comment (takagi) [inline tests/cupyx_tests/scipy_tests/linalg_tests/test_decomp_lu.py:36]: Would you remove this decorator from the helper method?

**Review by takagi — CHANGES_REQUESTED** [review]
LGTM except for one!

**Pull request closed without merging**